pkg/resource/deploy/plan_executor.go

```diff
@@ -213,6 +220,46 @@ func (pe *planExecutor) handleSingleEvent(event SourceEvent) *result.Result {
 	return nil
 }
 
+// retirePendingDeletes deletes all resources that are pending deletion. Run before the start of a plan, this pass
+// ensures that the engine never sees any resources that are pending deletion from a previous plan.
+//
+// retirePendingDeletes re-uses the plan executor's step generator but uses its own step executor.
+func (pe *planExecutor) retirePendingDeletes(callerCtx context.Context, opts Options, preview bool) error {
+	prev := pe.plan.prev
+	if prev == nil || len(prev.Resources) == 0 {
```

Comment:
Will `GeneratePendingDeletes` fail if there is no previous snapshot or `len(prev.Resources)` is 0?

Reply:
If not, we should remove this logic.

Reply:
Nope, it should be fine. I'll get rid of this.